### ISSUE TITLE
Output redirects as JSON or other structured data, to be used as an input to find/replace tools.

### ISSUE THREAD

**Issue by adarsh** [endymion/link-checker#7]
Great gem - really awesome and glad it's so well tested.

For the yellow redirect notices, it would be cool to pass in a `--replace` flag and have those url's just get updated in-place. Would save tons of tedious fixing.

Thanks for maintaining!

Edit: Updated title of issue to reflect new direction.

**Comment by mttrb** [comment]
The link-checker was originally created to check links in Octopress blogs.  Octopress is a static blog generator tool.  The problem this presents is that the files that are being checked are not the original source files.  I suppose it would be possible to have the link-checker update the source files (rather than the generated HTML files) but then the link-checker would probably need to cause Octopress to regenerate the blog.  Obviously this only applies if you are using the link-checker to check links in Octopress blogs.  If you added this functionality for Octopress then I suspect people would request similar functionality for other environments. This would quickly make the link-checker bloated and unwieldy.

A potential, neater solution might be to have the link-checker produce a machine readable report of the redirects and broken links and then have a separate product to fix the links based on the report, link-fixer.  You could then have environment specific link-fixers that understand Octopress blogs etc.

Anyway, just some thoughts.

**Comment by adarsh** [comment]
Yeah good idea. I'm using it for Markdown, but still works as a concept.

Something like JSON output would be useful and could be used as an input to any number of separate find/replace tools [(aside: our bin script for finding/replacing using silver searcher)](https://github.com/thoughtbot/dotfiles/blob/master/bin/replace).
